### PR TITLE
Validator for callables as parameter

### PR DESCRIPTION
**Pull request by AdriaanRol**
Changes proposed in this pull request:
- A validator for callable objects
- includes tests 

@giulioungaretti

### CODE REVIEW

File: qcodes/utils/validators.py [hunk 1]
```diff
@@ -420,3 +420,20 @@ def validate(self, value, context=''):
...
+class Callable(Validator):
+    """
+    Validator for callables such as functions.
+    """
+    def __init__(self):
+        # exists only to overwrite parent class
+        pass
+
+    def validate(self, value, context=''):
+        if not hasattr(value, '__call__'):
```
Comment (jenshnielsen) [inline qcodes/utils/validators.py]: Any reason to not use `if not callable(value)`

**Comment by jenshnielsen** [comment]
Looks good to me apart from a single inline comment

File: qcodes/utils/validators.py (hunk 1 above)
Comment (AdriaanRol) [inline qcodes/utils/validators.py]: Good suggestion, I'll make the change.

**Comment by AdriaanRol** [comment]
@jenshnielsen , I addressed your comments, let me know what you think.

**Comment by jenshnielsen** [comment]
@AdriaanRol Great looks good to me

**Pull request merged**